Volumes did not mount correctly when Docker bridge had IPv6 enabled

thallgren opened the pull request:
Telepresence failed to mount volumes after connecting with `telepresence
connect --docker` when Docker Engine had IPv6 enabled in its default
bridge network. Disabling IPv6 in the Telepresence client configuration
did not resolve the issue. This was fixed in Telepresence Volume Plugin
"telemount" version 0.3.2, which circumvented a
[bug in sshfs](https://github.com/libfuse/sshfs/issues/335).
Additionally, the volume plugin will no longer use IPv6 when the client
configuration `docker.enableIPv6` is set to `false`.

bgruszka committed:
Fix linter problems reported by golangci-lint 2.6.1

Signed-off-by: Błażej Gruszka <blazej.gruszka@displate.com>
Signed-off-by: Thomas Hallgren <thomas@tada.se>

thallgren committed:
Update module dependencies.

Signed-off-by: Thomas Hallgren <thomas@tada.se>

thallgren committed:
Volumes did not mount correctly when Docker bridge had IPv6 enabled

Telepresence failed to mount volumes after connecting with `telepresence
connect --docker` when Docker Engine had IPv6 enabled in its default
bridge network. Disabling IPv6 in the Telepresence client configuration
did not resolve the issue. This was fixed in Telepresence Volume Plugin
"telemount" version 0.3.2, which circumvented a
[bug in sshfs](https://github.com/libfuse/sshfs/issues/335).
Additionally, the volume plugin will no longer use IPv6 when the client
configuration `docker.enableIPv6` is set to `false`.

Signed-off-by: Thomas Hallgren <thomas@tada.se>

thallgren commented on pkg/client/docker/daemon.go:
```diff
@@ -330,7 +338,7 @@ func enableK8SAuthenticator(ctx context.Context, daemonID *daemon.Identifier) er
 			cfg := client.GetConfig(ctx).Docker()
 			kubeAuthHost := cfg.HostGateway
-			if !(cfg.AddHostGateway || kubeAuthHost != client.DefaultHostGateway) {
+			if kubeAuthHost == "" {
```
The `cfg.AddHostGateway` was incorrectly used here. It's controlling whether the host should be added when the daemon is started (it must be added on Linux), not if it can be used once it's up and running.

thallgren commented:
@bgruszka I cherry-picked one of your commits that fixed a lot of linter problems and modified it slightly. I'd like this fix to go into a 2.25.1 that we release shortly.

bgruszka reviewed (approved):
Looks good to me :) 

Only question I have is about this:

What will happen when both EnableIPv4: true and EnableIPv6: true ? Looking at free_ports.go, it chooses tcp4 by default, only using tcp6 when enableIPv6 is true

Should it probably check if enableIPv6 && !enableIPv4 or have a priority preference documented?

thallgren committed:
Disable docker.EnableIPv6 unless the bridge network enables it

Replaced multiple `docker.EnableClient` with a more streamlined approach
and removed unnecessary calls.

Signed-off-by: Thomas Hallgren <thomas@tada.se>

thallgren commented:
@bgruszka I changed the `FreePortsTCP` so that it just does `net.Listen("tcp", ":0")`. I found that it doesn't really matter as long as we don't pass the actual IP to Docker.
